src/triage/triage.controller.ts

```diff
@@ -1,13 +1,90 @@
-import { Body, Controller, Post } from '@nestjs/common';
+/* eslint-disable prettier/prettier */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { Body, Controller, Get, Param, Patch, Post, Req, UseGuards, Query } from '@nestjs/common';
 import { TriageService } from './triage.service';
 import { SubmitAlertDto } from './dto/submit-alert.dto';
+import { UpdateAlertDto } from './dto/update-alert.dto';
+import { ConvertAlertToCase } from './dto/convert-alert-to-case.dto';
+import { CloseAlertDto } from './dto/close-alert.dto';
+import { AuthGuard } from '@nestjs/passport';
+import { Roles } from '../auth/roles.decorator';
+import { RolesGuard } from '../auth/roles.guard';
 
 @Controller('api/v1/triage/alerts')
+@UseGuards(AuthGuard('jwt'), RolesGuard)
 export class TriageController {
   constructor(private readonly triageService: TriageService) {}
 
-  @Post()
-  async submitAlert(@Body() submitAlertDto: SubmitAlertDto) {
-    return this.triageService.handleAlert(submitAlertDto);
+  @Post('')
+  @Roles('CMS-TEST-ROLE', 'manage-account')
+  async submitAlert(@Body() dto: SubmitAlertDto, @Req() req) {
+    const userId = req.user.user_id;
+    const tenantId = req.user.tenantId;
+
+    return await this.triageService.handleNewAlert(dto, userId, tenantId, 'REST API');
+  }
```

Comment:
Example of how a controller handler should be handled:
```suggestion
  @Post('')
  @RequireCMSTestRole()
  async submitAlert(@Body() dto: SubmitAlertDto, @Req() req: AuthenticatedRequest) {
    const userId = req.user.token.clientId;
    const tenantId = req.user.token.tenantId;

    return await this.triageService.handleNewAlert(dto, userId, tenantId, 'REST API');
  }
```